### PR TITLE
Feat/10 daily agenda

### LINKED ISSUE #10: [FEAT] 오늘의 아젠다 기능 구현
## ✨작성할 기능에 대한 설명을 적어주세요
```
오늘의 아젠다 기능을 구현해야 한다.
기본적으로 Todo의 기능을 제공한다. (CRUD필요)
미완료 아젠다의 경우 다음날로 이동한다.
동시에 지나간 일자의 아젠다는 조회만 가능하고 수정은 불가능하다. (프론트와 추후 협의)
사용자가 드로그 앤 드랍으로 바꾸는 아젠다 리스트의 인덱싱을 처리해야 한다.
아젠다 해당달의 아젠다 수행률에 대한 로직을 계산하여 반환해줘야 한다.
```

<br><br>

## ✅ 해야 할 작업들을 작성해주세요
- [x] DailyAgenda CRUD 구현
- [x] 아젠다 수행률 로직 구현
- [x] 미완료 아젠다 처리 기능 구현
- [x] 아젠다 정렬 및 인덱싱 기능 구현
- [x] 에러처리
- [x] Test코드 작성

<br><br>

## 🦋 연관된 이슈
-

### LINKED ISSUE #18: [FEAT] 아젠다 pagingId 별 정렬하는 bulkupdate 쿼리 작성
## ✨작성할 기능에 대한 설명을 적어주세요
```
[FEAT] 아젠다 pagingId 별 정렬하는 bulkupdate 쿼리 작성
```

<br><br>

## ✅ 해야 할 작업들을 작성해주세요
- [x] bulkupdate 쿼리 작성
- [x] 테스트코드 작성

<br><br>

## 🦋 연관된 이슈
-

### PR DESCRIPTION
**Pull request by OneK-2**
## ✨ 요약
```
dailyAgenda의 crud 구현
미완료 아젠다의 경우 다음날로 복사하는 기능 구현
오늘의 아젠다 정렬 기능
아젠다 수행률 반환 기능 
```

<br><br>

## 😎 해결한 이슈
- close #10 

<br><br>

### CODE REVIEW

File: src/main/java/com/example/rework/dailyagenda/application/ScheduleService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                DailyAgenda newAgenda = DailyAgenda.builder()
+                        .member(agenda.getMember())
+                        .id(agenda.getId())
+                        .state(agenda.isState())
+                        .todo(agenda.getTodo())
+                        .build();
+                result.add(newAgenda);
+            }
+        }
+        dailyAgendaRepository.saveAll(result);
```
Comment (minwoo1999) [inline src/main/java/com/example/rework/dailyagenda/application/ScheduleService.java:52]: 테스트하기 편하게 saveAll로 처리하는부분 너무 좋은 것 같습니다~

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [84%]|:green_apple:|
|:-|:-:|:-:|
|[MonthlyAgendaController.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fpresentation%2FMonthlyAgendaController.java)|100%|:green_apple:|
|[DailyAgendaController.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fpresentation%2FDailyAgendaController.java)|100%|:green_apple:|
|[ScheduleService.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2FScheduleService.java)|100%|:green_apple:|
|[DailyAgendaServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fimpl%2FDailyAgendaServiceImpl.java)|89.25%|:green_apple:|
|[DailyAgenda.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fdomain%2FDailyAgenda.java)|88.68%|:green_apple:|
|[MonthlyAgendaRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fdto%2FMonthlyAgendaRequestDto.java)|85.93%|:green_apple:|
|[DailyAgendaRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fdto%2FDailyAgendaRequestDto.java)|85.65%|:green_apple:|
|[MonthlyAgendaServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fimpl%2FMonthlyAgendaServiceImpl.java)|83.27%|:green_apple:|
|[MonthlyAgenda.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fdomain%2FMonthlyAgenda.java)|81.93%|:green_apple:|
|[MonthlyAgendaResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fdto%2FMonthlyAgendaResponseDto.java)|81.9%|:green_apple:|
|[DailyAgendaResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fdto%2FDailyAgendaResponseDto.java)|61.11%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/81dfab2271ec3eddd1257d82e0d0751a6e917a17/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|

|Total Project Coverage|74.17%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/example/rework/dailyagenda/application/impl/DailyAgendaServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        return ReadDailyCompleteRateResponseDto.builder()
+                .completeCount(count)
+                .allCount(length)
+                .build();
+
+    }
+
+    @Override
+    public ReadMonthlyCompleteRateResponseDto readMonthlyCompleteRate(ReadMonthlyCompleteRateRequestDto readMonthlyCompleteRateRequestDto, SecurityUtils securityUtils) {
+        System.out.println("test123");
```
Comment (minwoo1999) [inline src/main/java/com/example/rework/dailyagenda/application/impl/DailyAgendaServiceImpl.java]: 오잉 이게 왜 들어가 있지 ...? ㅋㅋ 지워주실 수 있을까요 형님

**Review by minwoo1999 — APPROVED** [review]
고생많으셨습니다 형님 !! 나중에 testcode 리팩토링할 때 코드너리에서 본 것 같은데 assertion를 여러번 쓰는거보다 한번에 검증하는게 좋다고 하더라고요 나중에 시간될 때 assertionAll로 한번에 다 검증하는 걸로 바꿔보면 좋을 것 같아요!!

File: src/main/java/com/example/rework/dailyagenda/application/ScheduleService.java (hunk 1 above)
Comment (OneK-2) [reply src/main/java/com/example/rework/dailyagenda/application/ScheduleService.java:52]: 좋습니다~
리팩토링 리스트 같은거 미리 까먹기전에 미리 정리 좀 해놔야겠넴

File: src/main/java/com/example/rework/dailyagenda/application/impl/DailyAgendaServiceImpl.java (hunk 2 above)
Comment (OneK-2) [reply src/main/java/com/example/rework/dailyagenda/application/impl/DailyAgendaServiceImpl.java]: 앗 못봣네 ㅋㅋㅋㅋ 네넵

**Comment by minwoo1999** [comment]
### 테스트 커버리지 측정
|File|Coverage [83.98%]|:green_apple:|
|:-|:-:|:-:|
|[MonthlyAgendaController.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fpresentation%2FMonthlyAgendaController.java)|100%|:green_apple:|
|[DailyAgendaController.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fpresentation%2FDailyAgendaController.java)|100%|:green_apple:|
|[ScheduleService.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2FScheduleService.java)|100%|:green_apple:|
|[DailyAgendaServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fimpl%2FDailyAgendaServiceImpl.java)|89.2%|:green_apple:|
|[DailyAgenda.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fdomain%2FDailyAgenda.java)|88.68%|:green_apple:|
|[MonthlyAgendaRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fdto%2FMonthlyAgendaRequestDto.java)|85.93%|:green_apple:|
|[DailyAgendaRequestDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fdto%2FDailyAgendaRequestDto.java)|85.65%|:green_apple:|
|[MonthlyAgendaServiceImpl.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fimpl%2FMonthlyAgendaServiceImpl.java)|83.27%|:green_apple:|
|[MonthlyAgenda.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fdomain%2FMonthlyAgenda.java)|81.93%|:green_apple:|
|[MonthlyAgendaResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fmonthlyagenda%2Fapplication%2Fdto%2FMonthlyAgendaResponseDto.java)|81.9%|:green_apple:|
|[DailyAgendaResponseDto.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2Fdailyagenda%2Fapplication%2Fdto%2FDailyAgendaResponseDto.java)|61.11%|:green_apple:|
|[ReworkApplication.java](https://github.com/rework-kr/REWORK-SERVER/blob/4c86c7b137c022ea57a06d269b05359842d5f372/src%2Fmain%2Fjava%2Fcom%2Fexample%2Frework%2FReworkApplication.java)|37.5%|:green_apple:|

|Total Project Coverage|74.16%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**